File: kelmah-mobile-android/app/src/main/java/com/kelmah/mobile/features/profile/data/ProfileModels.kt
package com.kelmah.mobile.features.profile.data

data class WorkerProfileSnapshot(
    val profile: WorkerRecommendationProfile = WorkerRecommendationProfile(),
    val credentials: WorkerCredentials = WorkerCredentials(),
    val availability: WorkerAvailability = WorkerAvailability(),
    val completeness: WorkerCompleteness = WorkerCompleteness(),
    val portfolio: WorkerPortfolio = WorkerPortfolio(),
) {
    val visibleSkills: List<String>
        get() = credentials.skills.map { it.name }.ifEmpty { profile.skills }
}

data class WorkerRecommendationProfile(
    val id: String? = null,
    val firstName: String = "",
    val lastName: String = "",
    val email: String = "",
    val phone: String = "",
    val bio: String = "",
    val location: String = "",
    val profession: String = "",
    val hourlyRate: Double? = null,
    val currency: String = "GHS",
    val experienceLevel: String? = null,
    val yearsOfExperience: Int? = null,
    val skills: List<String> = emptyList(),
    val isEmailVerified: Boolean = false,
    val isPhoneVerified: Boolean = false,
) {
    val displayName: String
        get() = listOf(firstName, lastName)
            .filter { it.isNotBlank() }
            .joinToString(" ")
            .ifBlank { email.ifBlank { "Kelmah Worker" } }
}

data class WorkerCredentials(
    val skills: List<CredentialSkill> = emptyList(),
    val licenses: List<CredentialItem> = emptyList(),
    val certifications: List<CredentialItem> = emptyList(),
)

data class CredentialSkill(
    val id: String,
    val name: String,
    val category: String = "general",
    val proficiencyLevel: String = "intermediate",
    val yearsOfExperience: Int = 0,
    val isVerified: Boolean = false,
)

data class CredentialItem(
    val id: String,
    val name: String,
    val issuingOrganization: String = "",
    val issueDate: String? = null,
    val expiryDate: String? = null,
    val status: String = "pending",
    val isVerified: Boolean = false,
)

data class WorkerAvailability(
    val status: String = "not_set",
    val isAvailable: Boolean = true,
    val timezone: String = "Africa/Accra",
    val schedule: List<AvailabilityDay> = emptyList(),
    val nextAvailable: String? = null,
    val lastUpdated: String? = null,
    val message: String? = null,
)

data class AvailabilityDay(
    val day: String,
    val available: Boolean,
    val slots: List<AvailabilitySlot> = emptyList(),
)

data class AvailabilitySlot(
    val start: String,
    val end: String,
)

data class WorkerCompleteness(
    val completionPercentage: Int = 0,
    val requiredCompletion: Int = 0,
    val optionalCompletion: Int = 0,
    val missingRequired: List<String> = emptyList(),
    val missingOptional: List<String> = emptyList(),
    val recommendations: List<String> = emptyList(),
)

data class WorkerPortfolio(
    val items: List<PortfolioProject> = emptyList(),
    val totalCount: Int = 0,
    val publishedCount: Int = 0,
)

data class PortfolioProject(
    val id: String,
    val title: String,
    val description: String = "",
    val projectType: String = "professional",
    val skillsUsed: List<String> = emptyList(),
    val location: String? = null,
    val clientRating: Double? = null,
    val status: String = "draft",
    val isFeatured: Boolean = false,
    val createdAt: String? = null,
)

File: kelmah-mobile-ios/Kelmah/Features/Profile/Data/ProfileModels.swift
import Foundation

struct WorkerProfileSnapshot: Hashable {
    let profile: WorkerRecommendationProfile
    let credentials: WorkerCredentials
    let availability: WorkerAvailability
    let completeness: WorkerCompleteness
    let portfolio: WorkerPortfolio
    let partialWarnings: [String]

    var visibleSkills: [String] {
        let credentialSkills = credentials.skills.map(\.name)
        return credentialSkills.isEmpty ? profile.skills : credentialSkills
    }
}

struct WorkerRecommendationProfile: Hashable {
    let id: String?
    let firstName: String
    let lastName: String
    let email: String
    let phone: String
    let bio: String
    let location: String
    let profession: String
    let hourlyRate: Double?
    let currency: String
    let experienceLevel: String?
    let yearsOfExperience: Int?
    let skills: [String]
    let isEmailVerified: Bool
    let isPhoneVerified: Bool

    var displayName: String {
        [firstName, lastName]
            .filter { $0.isEmpty == false }
            .joined(separator: " ")
            .trimmingCharacters(in: .whitespacesAndNewlines)
            .nilIfEmpty ?? email.nilIfEmpty ?? "Kelmah Worker"
    }
}

struct WorkerCredentials: Hashable {
    let skills: [CredentialSkill]
    let licenses: [CredentialItem]
    let certifications: [CredentialItem]

    init(
        skills: [CredentialSkill] = [],
        licenses: [CredentialItem] = [],
        certifications: [CredentialItem] = []
    ) {
        self.skills = skills
        self.licenses = licenses
        self.certifications = certifications
    }
}

struct CredentialSkill: Hashable, Identifiable {
    let id: String
    let name: String
    let category: String
    let proficiencyLevel: String
    let yearsOfExperience: Int
    let isVerified: Bool
}

struct CredentialItem: Hashable, Identifiable {
    let id: String
    let name: String
    let issuingOrganization: String
    let issueDate: String?
    let expiryDate: String?
    let status: String
    let isVerified: Bool
}

struct WorkerAvailability: Hashable {
    let status: String
    let isAvailable: Bool
    let timezone: String
    let schedule: [AvailabilityDay]
    let nextAvailable: String?
    let lastUpdated: String?
    let message: String?

    init(
        status: String = "not_set",
        isAvailable: Bool = true,
        timezone: String = "Africa/Accra",
        schedule: [AvailabilityDay] = [],
        nextAvailable: String? = nil,
        lastUpdated: String? = nil,
        message: String? = nil
    ) {
        self.status = status
        self.isAvailable = isAvailable
        self.timezone = timezone
        self.schedule = schedule
        self.nextAvailable = nextAvailable
        self.lastUpdated = lastUpdated
        self.message = message
    }
}

struct AvailabilityDay: Hashable, Identifiable {
    let day: String
    let available: Bool
    let slots: [AvailabilitySlot]

    var id: String { day }
}

struct AvailabilitySlot: Hashable {
    let start: String
    let end: String
}

struct WorkerCompleteness: Hashable {
    let completionPercentage: Int
    let requiredCompletion: Int
    let optionalCompletion: Int
    let missingRequired: [String]
    let missingOptional: [String]
    let recommendations: [String]

    init(
        completionPercentage: Int = 0,
        requiredCompletion: Int = 0,
        optionalCompletion: Int = 0,
        missingRequired: [String] = [],
        missingOptional: [String] = [],
        recommendations: [String] = []
    ) {
        self.completionPercentage = completionPercentage
        self.requiredCompletion = requiredCompletion
        self.optionalCompletion = optionalCompletion
        self.missingRequired = missingRequired
        self.missingOptional = missingOptional
        self.recommendations = recommendations
    }
}

struct WorkerPortfolio: Hashable {
    let items: [PortfolioProject]
    let totalCount: Int
    let publishedCount: Int

    init(items: [PortfolioProject] = [], totalCount: Int = 0, publishedCount: Int = 0) {
        self.items = items
        self.totalCount = totalCount
        self.publishedCount = publishedCount
    }
}

struct PortfolioProject: Hashable, Identifiable {
    let id: String
    let title: String
    let description: String
    let projectType: String
    let skillsUsed: [String]
    let location: String?
    let clientRating: Double?
    let status: String
    let isFeatured: Bool
    let createdAt: String?
}

struct ProfileRawEnvelope: Decodable {
    let success: Bool?
    let message: String?
    let data: JSONValue?
    let meta: JSONValue?
}